### PR TITLE
batches: support new version field

### PR DESCRIPTION
**Pull request by stefanhengl**
This bumps sourcegraph/sourcegraph/lib to pull in the latest changes from https://github.com/sourcegraph/sourcegraph/pull/63613 to support the new `version` field for batch specs.

Notes:
 - To resolve repos, batches calls the GraphQL endpoint `resolveWorkspacesForBatchSpec`, which takes the serialized spec. This means all the logic to resolve repos is outsourced to Sourcegraph. This is great news for this PR because we don't have to worry how `on.RepositoriesMatchingQuery` is interpreted. The code path in which src-cli itself resolves the workspaces was removed in #819.
 
### Test plan
- New unit tests
- Manual testing. I tested the following workflow
  - `src batch new -f test.spec.yaml` (-> spec contains `version: 2`)
  - `src batch apply -f test.spec.yaml`   (-> validation passes,  batch spec shows up in Sourcegraph db)
  - `src batch validate -f test.spec.yaml` (-> validation passes or fails if I change the version to an unsupported value)
  - `src batch repos -f test.spec.yaml` (-> checked that returned repos match with expected pattern type) 

Closes SPLF-126

### CODE REVIEW

File: internal/batches/docker/image.go [hunk 1]
```diff
@@ -103,7 +103,7 @@ func (image *image) Ensure(ctx context.Context) error {
 			// docker image inspect will return a non-zero exit code if the image and
 			// tag don't exist locally, regardless of the format.
 			var digest string
-			if digest, err = inspectDigest(); errors.HasType(err, &fastCommandTimeoutError{}) {
+			if digest, err = inspectDigest(); errors.HasType[*fastCommandTimeoutError](err) {
```
Comment (stefanhengl) [inline internal/batches/docker/image.go:106]: `errors.HasType` was changed in Sourcegraph so I had to update the call here. (See https://github.com/sourcegraph/sourcegraph/pull/63024)

File: internal/batches/service/service.go [hunk 2]
```diff
@@ -506,7 +506,9 @@ func validateMount(batchSpecDir string, spec *batcheslib.BatchSpec) error {
 	return nil
 }
 
-const exampleSpecTmpl = `name: NAME-OF-YOUR-BATCH-CHANGE
+const exampleSpecTmpl = `# "version" determines how the batch spec is interpreted.
+version: 2
+name: NAME-OF-YOUR-BATCH-CHANGE
```
Comment (stefanhengl) [inline internal/batches/service/service.go:511]: This is returned by `src batch new -f <file>`

Comment (jtibshirani) [reply internal/batches/service/service.go:511]: Tiny comment, as a user I would find `# "version" determines how the batch spec is interpreted` to be a bit mysterious. Maybe we could say
```
version: 2 # Use the latest schema version
name: ...
```

**Review by jtibshirani — APPROVED** [review]
Looks great!

**Review by camdencheek — APPROVED** [review]
Thanks!

**Pull request merged**